ocean/transects/compute_transects.py

```diff
@@ -31,41 +32,111 @@ def get_mask_short_names(mask):
 
+  edgesToRead = edgeVals[0,:nEdgesInTransect[0]]
+  for i in range(1,nTransects):
+    edgesToRead = np.hstack([edgesToRead,edgeVals[i,:nEdgesInTransect[i]]])
+
+  edgesToRead = np.asarray(edgesToRead, dtype='i')
+
+  dvEdge = mesh.dvEdge.sel(nEdges=edgesToRead).values
+
+  edgeSigns = np.zeros((nTransects,len(edgesToRead)))
+  for i in range(nTransects):
+    edgeSigns[i,:] = mask.sel(nEdges=edgesToRead, shortNames=transectList[i]).squeeze().transectEdgeMaskSigns.values
+
+# Read time average files one at a time and slice
+  fileList = sorted(glob.glob(timeavg))
+  transport = np.zeros((len(fileList),nTransects))
+  t = np.zeros(len(fileList))
+  for i,fname in enumerate(fileList):
+    ncid = Dataset(fname,'r')
+    vel = ncid.variables['timeMonthly_avg_normalTransportVelocity'][0,edgesToRead,:]
+    t[i] = ncid.variables['timeMonthly_avg_daysSinceStartOfSim'][:] / 365.
+    ncid.close()
+#   Compute transport for each transect
+    for j in range(nTransects):
+      start = int(nTransectStartStop[j])
+      stop = int(nTransectStartStop[j+1])
+      transport[i,j] = (dvEdge[start:stop,np.newaxis]*h[np.newaxis,:]*vel[start:stop,:] \
+          *edgeSigns[j,start:stop,np.newaxis]).sum()*m3ps_to_Sv
+
+    print fname
+ 
+  for j in range(nTransects):
+    plt.figure()
+    plt.plot(t, transport[:,j], 'k-', lw=3, label='monthly-averaged transport')
+    if transectList[j] == 'Drake Passage':
+      plt.gca().fill_between(t, (134.-14.)*np.ones_like(t), (134.+14.)*np.ones_like(t), alpha=0.3, label='observations')
+    plt.legend(loc='best', frameon=False)
+    plt.ylabel('Transport (Sv)')
+    plt.xlabel('yrs')
+    plt.title('Transport for ' + name)
+    plt.savefig('transport' + transectList[j].replace(' ', '') +'.png')
+
+# Add calls to save transport and then can build up
+  ncid=Dataset('transport.nc',mode='w',clobber=True, format='NETCDF3_CLASSIC')
+  ncid.createDimension('Time',None)
+  ncid.createDimension('nTransects',nTransects)
+  ncid.createDimension('StrLen',64)
+  transectNames=ncid.createVariable('TransectNames','c',('nTransects','StrLen'))
+  times=ncid.createVariable('Time','f8','Time')
+  transportOut=ncid.createVariable('Transport','f8',('Time','nTransects'))
+ 
+  times[:] = t
+  transportOut[:,:] = transport
+
+  for i in range(nTransects):
+    nLetters = len(transectList[i])
+    transectNames[i,:nLetters] = transectList[i]
+  ncid.close() 
+  
 
 if __name__ == "__main__":
   import argparse
   parser = argparse.ArgumentParser(description=__doc__,
                                    formatter_class=argparse.RawTextHelpFormatter)
-  parser.add_argument("-o", "--output_file_pattern", dest="output_filename_pattern",
-      help="MPAS Filename pattern for output.", metavar="FILE",
-      required=True)
+#  parser.add_argument("-o", "--output_file_pattern", dest="output_filename_pattern",
+#      help="MPAS Filename pattern for output.", metavar="FILE",
+#      required=True)
```

Comment:
I think you should just remove this text here if you decide to remove the output file name.

Reply:
done